api/client/crop_model.py

```diff
@@ -69,3 +70,75 @@ def compute_crop_weighted_series(self,
             # TODO: change metric to reflect it is weighted in this copy
             series_list.append(series)
         return pandas.concat(series_list)
+
+
+    def growing_degree_days(self, region_name, base_temperature,
+                            start_date, end_date, min_temporal_coverage=0.5):
+        """Get Growing Degree Days (GDD) for a region.
+
+        Growing degree days (GDD) are a weather-based indicator that
+        allows for assessing crop phenology and crop development,
+        based on heat accumulation. GDD for one day is defined as
+        [T_mean - T_base], where T_mean is the average temperature of
+        that day if available. Typically T_mean is approximated as
+        (T_max + T_min)/2.
+
+        The GDD over a longer time interval is the sum of the GDD over
+        all days in the interval. Days where the data is missing
+        contribute 0 GDDs, i.e. are treated as if T_mean = T_base.
+        Use the temporal coverage threshold to avoid computing GDD
+        with too little data.
+
+        The threshold and the base temperature should be carefuly
+        selected based on fundamental understanding of the crops and
+        region of interest.
+
+        The region can be any region of the Gro regions, from a point
+        location to a district, province etc. This will use the best
+        available data series for T_max and T_min for the given region
+        and time period, using "find_data_series". 
+
+        In the simplest case, if the given region is a weather station
+        location which has data for the time period, then that will be
+        used. If it's a district or other region, the underlying data
+        could be from one or more weather stations and/or satellite.
+
+        Parameters
+        ----------
+        region_name: string, required
+        base_temperature: number, required
+        start_date: '%Y-%m-%d' string, required
+        end_date: '%Y-%m-%d' string, required
+        min_temporal_coverage: float, optional
+
+        """
+        for tmax in self.find_data_series(
```

Comment:
Since we are not specifying the source_id here, it could be LST or GHCN or some other future sources right?

Reply:
Right. It will use  the source ranking for that time range and region.

Reply:
In that case, a user would bear the risk of having their daily mean temperature calculated from mixed sources. The risk being that certain regions or dates with more availability of one source have a more biased temperature than others, since LST tends to have higher Tmean than the GHCN Tmean. I think the current script is fine if a user can ignore such biases or uncertainties attributed to various data availability scenarios. Meanwhile, a user should be able to specify one particular source to use exclusively as well, like an argument `source_name` or `source_id` with default value Null, in which case the source ranking decides the source(s).

Reply:
Right.. 

growing_degree_days()  uses find_data_series()  entity search and series availability  to  avoid the user having to know which specific series they need, which depends on the region and the date ranges. 

I've separated the lower level calculation into a separate function compute_gdd() so a user who already knows how to select specific series can do it directly.